src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -122,12 +122,16 @@ internal unsafe SocketError DoOperationConnect(Socket socket, SafeCloseSocket ha
 
         internal SocketError DoOperationDisconnect(Socket socket, SafeCloseSocket handle)
         {
-            throw new PlatformNotSupportedException(SR.net_sockets_disconnect_notsupported);
+            SocketError socketError = SocketPal.Disconnect(socket, handle, _disconnectReuseSocket);
+            if (socketError != SocketError.IOPending)
+            {
+                FinishOperationSync(socketError, 0, SocketFlags.None);
+            }
```

Comment:
And what if it does == IOPending?  Or maybe given the implementation, we expect it can't ever be IOPending? In that case this should be changed to something like:
```C#
Debug.Assert(socketError != SocketError.IOPending, "Disconnect on Unix should never be pending");
FinishOperationSync(socketError, 0, SocketFlags.None);
return socketError;
```